corehq/apps/registration/tests/test_utils.py

```diff
@@ -58,3 +81,29 @@ def test_domain_is_not_active_for_new_user(self):
         )
         domain = Domain.get_by_name(domain_name)
         self.assertFalse(domain.is_active)
+
+    @mock.patch('corehq.apps.registration.utils._setup_subscription', _issue_initializing_domain)
+    @mock.patch('corehq.apps.registration.utils.notify_exception', _noop)
+    def test_subscription_exception_raises_error(self):
+        with self.assertRaisesMessage(ErrorInitializingDomain,
+                                      "Subscription setup failed for 'subscription-failed'"):
+            request_new_domain(
+                self.request,
+                'subscription-failed',
+                is_new_user=True,
+            )
+        domain = Domain.get_by_name('subscription-failed')
+        self.assertIsNone(domain)
+
+    @mock.patch('corehq.apps.registration.utils.initialize_domain_with_default_roles', _issue_initializing_domain)
+    @mock.patch('corehq.apps.registration.utils.notify_exception', _noop)
+    def test_default_roles_exception_raises_error(self):
+        with self.assertRaisesMessage(ErrorInitializingDomain,
+                                      "Subscription setup failed for 'init-default-roles-failed'"):
```

Comment:
Does the error in the test output mean a `noop` needs to be added for `_setup_subscription` here?